File: algorithms/1_easy/M/01544_make_the_string_great/solution.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []

        for ch in s:
            if stack and (
                ord(stack[-1]) + 32 == ord(ch) or
                ord(ch) + 32 == ord(stack[-1])
            ):
                stack.pop()
            else:
                stack.append(ch)

        return "".join(stack)

File: algorithms/1_easy/M/01544_make_the_string_great/solution.rs
impl Solution {
    pub fn make_good(s: String) -> String {
        let mut stack: Vec<u8> = Vec::new();

        for &byte in s.as_bytes().iter() {
            if let Some(&top) = stack.last() && (
                top + b'A' - b'a' == byte ||
                byte + b'A' - b'a' == top
            ) {
                stack.pop();
            } else {
                stack.push(byte);
            }
        }

        String::from_utf8(stack).unwrap()
    }
}
